operators/pkg/controller/elasticsearch/certificates/http/reconcile.go

```diff
@@ -113,6 +107,51 @@ func reconcileHTTPInternalCertificateSecret(
 		secret.Data = make(map[string][]byte)
 	}
 
+	if customCertificates != nil {
+		if !reflect.DeepEqual(secret.Data, customCertificates.Data) {
+			needsUpdate = true
+			secret.Data = customCertificates.Data
+		}
+	} else {
+		selfSignedNeedsUpdate, err := verifyInternalSelfSignedCertificateSecret(
+			&secret, es, svcs, ca, certValidity, certReconcileBefore,
+		)
+		if err != nil {
+			return nil, err
+		}
+		needsUpdate = needsUpdate || selfSignedNeedsUpdate
+	}
+
+	if needsUpdate {
+		if shouldCreateSecret {
+			log.Info("Creating HTTP internal certificate secret", "secret", secret.Name)
+			if err := c.Create(&secret); err != nil {
+				return nil, err
+			}
+		} else {
+			log.Info("Updating HTTP internal certificate secret", "secret", secret.Name)
+			if err := c.Update(&secret); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	result := CertificatesSecret(secret)
+	return &result, nil
+}
+
+// verifyInternalSelfSignedCertificateSecret verifies the contents of a secret containing self-signed certificates.
+// Returns true if the secret needs to be updated.
+func verifyInternalSelfSignedCertificateSecret(
```

Comment:
IIUC it also updates `secret` in-place, which could be worth mentioning in the comment